crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -500,39 +476,53 @@ where
     ///         .with_executor(TokioTaskExecutor::default())
     ///         .with_events(TestCanonStateSubscriptions::default())
     ///         .with_evm_config(evm)
-    ///         .into_registry(Default::default());
+    ///         .into_registry(Default::default(), EthApiBuild);
     ///
     ///     let eth_api = registry.eth_api();
     /// }
     /// ```
-    pub fn into_registry(
+    pub fn into_registry<EthApi>(
         self,
         config: RpcModuleConfig,
-    ) -> RethModuleRegistry<Provider, Pool, Network, Tasks, Events, EvmConfig> {
+        eth: impl EthApiBuilder<Provider, Pool, EvmConfig, Network, Tasks, Events, Server = EthApi>
+            + 'static,
+    ) -> RpcRegistryInner<Provider, Pool, Network, Tasks, Events, EthApi>
+    where
+        EthApi: FullEthApiServer,
+    {
         let Self { provider, pool, network, executor, events, evm_config } = self;
-        RethModuleRegistry::new(provider, pool, network, executor, events, config, evm_config)
+        RpcRegistryInner::new(provider, pool, network, executor, events, config, evm_config, eth)
     }
 
     /// Configures all [`RpcModule`]s specific to the given [`TransportRpcModuleConfig`] which can
     /// be used to start the transport server(s).
     ///
     /// See also [`RpcServer::start`]
-    pub fn build(self, module_config: TransportRpcModuleConfig) -> TransportRpcModules<()> {
+    pub fn build<EthApi>(
```

Comment:
can we rename this to `build_with` and keep the existing `build` fn that calls `build_with` with the EthApiBuild?